sycl/test/separate-compile/cpp_version_mismatch_test_2.cpp

```diff
@@ -0,0 +1,63 @@
+// >> ---- device compilation
+// RUN: %clangxx -std=c++14 -fsycl-device-only -Xclang -fsycl-int-header=sycl_ihdr_a.h %s -c -o a_kernel.bc -I %sycl_include
+//
+// >> ---- host compilation: no cpp version mismatch
+// RUN: %clangxx -std=c++14 -include sycl_ihdr_a.h -c %s -I %sycl_include
+//
+// >> ---- code generation checks
+// RUN: FileCheck -input-file=sycl_ihdr_a.h %s
+// CHECK: #define STD_CPP_VERSION 
+// CHECK-NEXT: #if __cplusplus != STD_CPP_VERSION
+// CHECK-NEXT: #error "C++ version for host compilation does not match C++ version used for device compilation"
+// CHECK-NEXT: #endif
+
+//==----------- cpp_version_mismatch_test_2.cpp - SYCL separate compilation cpp version mismatch test -----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// -----------------------------------------------------------------------------
+#include <CL/sycl.hpp>
+#include <iostream>

```

Comment:
You don't need an kernel doing something for this test, do you? An empty kernel is sufficient.

```suggestion

namespace sycl = cl::sycl;

int main() {
    sycl::queue().submit([&](sycl::handler& cgh) {
        cgh.parallel_for<class task>(sycl::nd_range<1>(16,16),
            [=](auto item) {
            });});
}

BTW: Why don't we write out an empty integration header if `-fsycl-int-header` is provided but there is no kernel in the source file? I can see this causing an issue if users set up their build system but some files don't (yet) have a kernel. Than the required include file is missing.
```